File: src/backend/app/schemas/notification.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.enums import NotificationCategory


class NotificationCreate(BaseModel):
    user_id: uuid.UUID
    category: NotificationCategory
    title: str
    message: str
    force_send: bool = False


class NotificationRead(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    category: NotificationCategory
    title: str
    message: str
    is_read: bool
    telegram_sent: bool
    created_at: datetime
    model_config = {"from_attributes": True}


class UnreadCountRead(BaseModel):
    count: int

File: src/frontend/src/types/notification.ts
export type NotificationCategory = 'assessment' | 'idp' | 'career' | 'system'

export interface NotificationRead {
  id: string
  user_id: string
  category: NotificationCategory
  title: string
  message: string
  is_read: boolean
  telegram_sent: boolean
  created_at: string
}

export interface UnreadCountRead {
  count: number
}
